### PR TITLE
fix: prompt optimizations for create file/dir; address case of skipping prompt

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

- Fixed an issue where the location prompt for the `Create Directory` and `Create File` USS features would appear even when a path is already set for the profile or parent folder. [#3183](https://github.com/zowe/zowe-explorer-vscode/pull/3183)
- Fixed an issue where the `Create Directory` and `Create File` features would continue processing when the first prompt was dismissed, causing an incorrect URI to be generated. [#3183](https://github.com/zowe/zowe-explorer-vscode/pull/3183)
- Fixed an issue where the `Create Directory` and `Create File` features would incorrectly handle user-specified locations with trailing slashes. [#3183](https://github.com/zowe/zowe-explorer-vscode/pull/3183)

## Release Notes

Milestone: 3.0.1

Changelog: See `Proposed changes`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [x] These changes may need ported to the appropriate branches (list here): `v2-lts`

### CODE REVIEW

File: packages/zowe-explorer/src/trees/uss/USSActions.ts [hunk 1]
```diff
@@ -60,17 +54,26 @@ export class USSActions {
...
+            node.fullPath = filePath;
         } else {
             filePath = node.fullPath;
         }
+
+        if (filePath == null || filePath.length === 0) {
+            return;
+        }
+
+        const isTopLevel = SharedContext.isSession(node);
```
Comment (zFernand0) [inline packages/zowe-explorer/src/trees/uss/USSActions.ts:66]: would it make sense to move this above line 42 and reuse it (as with the v2 port)? 😋

File: packages/zowe-explorer/src/trees/uss/USSActions.ts [hunk 2]
```diff
@@ -82,6 +85,7 @@ export class USSActions {
                 } else {
```
Comment (zFernand0) [inline packages/zowe-explorer/src/trees/uss/USSActions.ts:85]: and sma equestion here... should we check for `node.fullPAth?.length === 0` here?

**Review by zFernand0 — APPROVED** [review]

File: packages/zowe-explorer/src/trees/uss/USSActions.ts (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer/src/trees/uss/USSActions.ts]: Yes, sorry I moved this in the v2 port and forgot to apply it here - addressed in 5ded04a

File: packages/zowe-explorer/src/trees/uss/USSActions.ts (hunk 2 above)
Comment (traeok) [reply packages/zowe-explorer/src/trees/uss/USSActions.ts:85]: We should be able to leave this as-is now that I moved up the `isTopLevel` variable 😋

**Review by zFernand0 — APPROVED** [review]

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @traeok

**Pull request merged**